wallet-new/integration/Functions.hs

```diff
@@ -548,26 +545,26 @@ runAction wc action = do
                 )
                 (UnexpectedChangeAddress changeWalletAddresses)
 
-            let checkWalletAddressAfter diffList expectedOperation = do
-                    log "checking expected addresses balances..."
-                    forM_ diffList $ \PaymentDistribution{..} -> do
-                        let mBeforePayment = find ((pdAddress ==) . addrId) addressesBeforeTransaction
-                            mAfterPayment = find ((pdAddress ==) . addrId) addressesAfterTransaction
-                        case (,) <$> mBeforePayment <*> mAfterPayment of
-                            Just (beforePayment, afterPayment) -> do
-                                let balanceBeforePaymentModified = V1 . expectedOperation (unV1 pdAmount) . unV1 $ addrBalance beforePayment
-                                    balanceAfterPayment = addrBalance afterPayment
-                                checkInvariant
-                                    (balanceBeforePaymentModified == balanceAfterPayment)
-                                    (UnexpectedAddressBalance beforePayment afterPayment)
-                            Nothing -> throwM $ CantFindAddress pdAddress
-
-            -- Check did addresses of all payment sources decrease by expected amount after transaction
-            checkWalletAddressAfter (txInputs newTx) $ flip unsafeSubCoin
-
-            -- Check did addresses of all payment destinations increase by expected amount after transaction
-            -- NOTE: we are using paymentDestinations instead of txOutputs to eliminate changeAddress
-            checkWalletAddressAfter paymentDestinations unsafeAddCoin
+
+            accountSourceAfter <-
```

Comment:
I reckon this is just formatting, right?

Reply:
Nope, that's actually the interesting part. The invariant before was checking that each addresses balance either decreased or increased according to the payment distribution. I replaced that to have the check done on the accounts' balances instead.